packages/x-archetype-utils/src/i18n/i18n.plugin.ts

```diff
@@ -0,0 +1,160 @@
+import { VueConstructor } from 'vue';
+import VueI18n, { LocaleMessageObject } from 'vue-i18n';
+import { deepMerge } from '@empathyco/x-deep-merge';
+import { AnyMessages, Device, I18nOptions, LoadLazyMessagesByDevice, Locale } from './i18n.types';
+
+/**
+ * I18n settings manager.
+ *
+ * @public
+ */
+export class I18n<SomeMessages> {
+  public vueI18n!: VueI18n;
+  protected locale!: Locale;
+  protected device!: Device;
+  protected messages!: Record<Locale, AnyMessages<SomeMessages>>;
+  protected currentMessages!: LocaleMessageObject;
+  protected fallbackLocale!: Locale;
+
+  /**
+   * Constructs a new {@link I18n} instance with the passed {@link I18nOptions | i18n options}.
+   *
+   * @param options - The new {@link I18nOptions | i18n options}.
+   */
+  public constructor({ locale, messages, device, fallbackLocale }: I18nOptions<SomeMessages>) {
+    this.locale = locale;
+    this.device = device;
+    this.messages = messages;
+    this.fallbackLocale = fallbackLocale;
+  }
+
+  /**
+   * Creates a new {@link I18n} instance.
+   *
+   * @param options - The new {@link I18nOptions | i18n options}.
+   *
+   * @returns The new instance.
+   */
+  static async create<SomeMessages>(
+    options: I18nOptions<SomeMessages>
+  ): Promise<I18n<SomeMessages>> {
+    const instance = new I18n(options);
+    instance.currentMessages = await instance.getCurrentMessages();
+    return instance;
+  }
+
+  /**
+   * Installs VueI18n plugin.
+   *
+   * @param vue - The Vue instance.
+   */
+  install(vue: VueConstructor): void {
+    vue.use(VueI18n);
+    this.vueI18n = new VueI18n({
+      locale: this.locale,
+      messages: this.currentMessages ? { [this.locale]: this.currentMessages } : {},
+      silentFallbackWarn: true,
+      missing: (locale, key) => {
+        return (
+          this.getMessageWithDotsInKey(key) ??
+          `[i18n] Key '${key}' is missing for locale: '${locale}'`
+        );
+      }
+    });
+  }
+
+  /**
+   * Tries to get a value from the messages object accessing it key by key.
+   * When a key fails, takes the remaining key parts and tries to access the message
+   * joining them into a single key.
+   *
+   * @param key - A dot separated key to access the message object.
+   * @returns The message corresponding to the provided key, or empty string if it doesn't exist.
+   */
+  protected getMessageWithDotsInKey(key: string): string {
+    const keyParts = key.split('.');
+    let keyPartMessages = this.currentMessages;
+
+    const index = keyParts.findIndex(keyPart => {
+      if (typeof keyPartMessages[keyPart] === 'object') {
+        keyPartMessages = <LocaleMessageObject>keyPartMessages[keyPart];
+      } else {
+        return true;
+      }
+    });
+
+    return index !== -1 ? <string>keyPartMessages?.[keyParts.slice(index).join('.')] : '';
+  }
```

Comment:
I would propose a little refactor with my favorite tool, the `reduce`.

-  I find quite weird mutating a variable outside the `findIndex` loop.
- This function should return a nullish value as per the usage above.
- `typeof keyPartMessages[keyPart]` can return `true` if the value is `null`. We should use `isObject`
- Messages can be an array or a function. I would clarify that we do not support this yet (I tried, but I think the benefit is not much and it complicates the code.

```ts
  /**
   * Tries to get a value from the messages object accessing it key by key.
   * When a key fails, takes the remaining key parts and tries to access the message
   * joining them into a single key.
   *
   * @param path - A dot separated key to access the message object.
   * @returns The message corresponding to the provided key, or empty string if it doesn't exist.
   */
  protected getMessageWithDotsInKey(path: string): string | null {
    // TO-DO Support function or array messages
    const message = path
      .split('.')
      .reduce<LocaleMessage>(
        (messages, key, index, pathParts) =>
          isObject(messages)
            ? messages[key] ?? messages[pathParts.slice(index).join('.')]
            : messages,
        this.currentMessages
      );
    return typeof message === 'string' ? message : null;
  }
```